tests/ui/while_pop_unwrap.rs

```diff
@@ -0,0 +1,47 @@
+#![allow(unused)]
+#![warn(clippy::while_pop_unwrap)]
+
+struct VecInStruct {
+    numbers: Vec<i32>,
+    unrelated: String,
+}
+
+fn accept_i32(_: i32) {}
+
+fn main() {
+    let mut numbers = vec![1, 2, 3, 4, 5];
+    while !numbers.is_empty() {
+        let number = numbers.pop().unwrap();
+    }
+
+    let mut val = VecInStruct {
+        numbers: vec![1, 2, 3, 4, 5],
+        unrelated: String::new(),
+    };
+    while !val.numbers.is_empty() {
+        let number = val.numbers.pop().unwrap();
+    }
+
+    while !numbers.is_empty() {
+        accept_i32(numbers.pop().unwrap());
+    }
+
+    while !numbers.is_empty() {
+        accept_i32(numbers.pop().expect(""));
+    }
+
+    // This should not warn. It "conditionally" pops elements.
+    while !numbers.is_empty() {
+        if true {
+            accept_i32(numbers.pop().unwrap());
+        }
+    }
+
+    // This should also not warn. It conditionally pops elements.
+    while !numbers.is_empty() {
+        if false {
+            continue;
+        }
+        accept_i32(numbers.pop().unwrap());
+    }
```

Comment:
I'd like to see a test that checks for `is_empty()` on a different `Vec`. Also one that `pop()`s, but doesn't `unwrap()`, and one macro-generated (can be with a local macro).

Reply:
Oh that's a good point. At the moment this lint doesn't actually check if `Vec::pop()` and `Vec::is_empty()` were called on the same `Vec`, but I'm also not sure how I'd check for that. Would it make sense to just compare the two `Expr`s (`SpanlessEq::eq_expr`?) or is there a better way?

Reply:
Yeah, `SpanlessEq` is the way to go there.